pkg/meta/redis/service/redis.go

```diff
@@ -163,57 +169,57 @@ func (c *client) Connect(ctx context.Context) error {
 }
 
 func (c *client) Get(key string) (string, error) {
-	return c.get(c.kvIndex, key)
+	return c.get(c.kvPrefix, key)
 }
 
 func (c *client) GetMultiple(keys ...string) (vals []string, err error) {
-	return c.getMulti(c.kvIndex, keys...)
+	return c.getMulti(c.kvPrefix, keys...)
 }
 
 func (c *client) GetInverse(val string) (string, error) {
-	return c.get(c.vkIndex, val)
+	return c.get(c.vkPrefix, val)
 }
 
 func (c *client) GetInverseMultiple(vals ...string) ([]string, error) {
-	return c.getMulti(c.vkIndex, vals...)
+	return c.getMulti(c.vkPrefix, vals...)
+}
+
+func appendPrefix(prefix, key string) string {
+	return prefix + "-" + key
 }
 
-func (c *client) get(idx int, key string) (string, error) {
+func (c *client) get(prefix, key string) (string, error) {
 	pipe := c.db.TxPipeline()
-	pipe.Select(idx)
-	res := pipe.Get(key)
+	res := pipe.Get(appendPrefix(prefix, key))
 	if _, err := pipe.Exec(); err != nil {
 		return "", err
 	}
 	return res.Result()
 }
 
-func (c *client) getMulti(idx int, keys ...string) (vals []string, err error) {
+func (c *client) getMulti(prefix string, keys ...string) (vals []string, err error) {
 	pipe := c.db.TxPipeline()
-	pipe.Select(idx)
-	res := pipe.MGet(keys...)
-	if _, err = pipe.Exec(); err != nil {
-		return nil, err
+	ress := make([]redis.StringCmd, 0, len(keys))
+	for _, k := range keys {
+		ress = append(ress, pipe.Get(appendPrefix(prefix, k)))
 	}
-	if err = res.Err(); err != nil {
+	if _, err = pipe.Exec(); err != nil {
 		return nil, err
 	}
-	vals = make([]string, 0, len(keys))
-	for _, r := range res.Val() {
-		val, ok := r.(string)
-		if ok {
-			vals = append(vals, val)
+	vals = make([]string, 0, len(ress))
+	for _, res := range ress {
+		if err = res.Err(); err != nil {
+			continue
```

Comment:
Should I use `return nil, err` here?

Reply:
Or revise the interface of this function to return `(vals []string, errs []error)`.